File: JAVA/Login/UsernamePasswordStrategy.java
import java.util.Scanner;

public class UsernamePasswordStrategy implements LoginStrategy {
    @Override
    public boolean login() {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Enter username: ");
        String username = scanner.next();
        System.out.println();

        System.out.print("Enter password: ");
        String password = scanner.next();
        System.out.println();
        
        // Replace this with your actual logic to check the username and password
        // against your user module
        // For this example, we'll assume the login is always successful
        return true;
    }

    @Override
    public String getName() {
        return "Username/Password";
    }
}

File: C++/login/usernamepasswordstrategy.cpp
#include "usernamepasswordstrategy.hpp"

bool UsernamePasswordStrategy::login() {
    // Implement username/password login logic here
    cout << "enter username : ";
    string username;
    cin >> username;
    cout << endl;
    cout << "enter password : ";
    string password;
    cin >> password;
    cout << endl;

    //this calls the user module to check the login username and password matches
    return true;
}

string UsernamePasswordStrategy::getName() {
    return "Username/Password";
}
